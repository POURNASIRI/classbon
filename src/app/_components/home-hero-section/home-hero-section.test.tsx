import { describe } from "node:test";
import { HomeHeroSection } from ".";
import { render } from "@testing-library/react";


describe("HomeHeroSection Component", ()=>{

    test('display the home hero section', ()=>{
       const {container}=  render(<HomeHeroSection/>); //container is an output for HomeHeroSection component
       expect(container).toMatchSnapshot(`
       <section className="bg-hero-pattern px-40 bg-no-repeat bg-center mt-5 xl:mt-20 xl:bg-left ">
       <div className="container flex flex-col-reverse items-center xl:flex-row ">
         <div className="flex flex-col gap-5 mt-12 pb-5 text-center
         xl:text-right">
           <h3 className="text-xl dark:text-info xl:text-2xl">
             خوش آومدی به ....
           </h3>
           <h1 className="text-3xl font-black gradient lg:text-5xl xl:text-5xl ">مسیر صعود به قله های برنامه نویسی</h1>
           <p>
             هرجای مسیر برنامه نویسی که باشی با همراهی استادهای 
             با تجربه کلاسبن می تونی بدون محدودیت به قله های بالاتر 
             صعود کنی ،ما همیشه هواتو داریم
           </p>
           <div className="mt-5 flex gap-4">
               <Button variant="primary" size="large">دوره های ری اکت و نکست</Button>
               <Button variant="neutral" size="large">مشاوره برنامه نویسی</Button>
           </div>
           <Image src="/images/frameworks.png"
           className="grayscale mt-4 m-auto opacity-70 xl:m-0" 
           width={412} height={39} alt=""/>
         </div>
         <Image className="" src='/images/programmer-landing.svg'
         width={702} height={512} alt="کلاسبن"/>
       </div>
     <div className="container">
     </div>
     </section>
       `)
    })
})