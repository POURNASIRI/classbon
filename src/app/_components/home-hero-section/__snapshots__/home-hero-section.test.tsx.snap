// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`display the home hero section: 
       <section className="bg-hero-pattern px-40 bg-no-repeat bg-center mt-5 xl:mt-20 xl:bg-left ">
       <div className="container flex flex-col-reverse items-center xl:flex-row ">
         <div className="flex flex-col gap-5 mt-12 pb-5 text-center
         xl:text-right">
           <h3 className="text-xl dark:text-info xl:text-2xl">
             خوش آومدی به ....
           </h3>
           <h1 className="text-3xl font-black gradient lg:text-5xl xl:text-5xl ">مسیر صعود به قله های برنامه نویسی</h1>
           <p>
             هرجای مسیر برنامه نویسی که باشی با همراهی استادهای 
             با تجربه کلاسبن می تونی بدون محدودیت به قله های بالاتر 
             صعود کنی ،ما همیشه هواتو داریم
           </p>
           <div className="mt-5 flex gap-4">
               <Button variant="primary" size="large">دوره های ری اکت و نکست</Button>
               <Button variant="neutral" size="large">مشاوره برنامه نویسی</Button>
           </div>
           <Image src="/images/frameworks.png"
           className="grayscale mt-4 m-auto opacity-70 xl:m-0" 
           width={412} height={39} alt=""/>
         </div>
         <Image className="" src='/images/programmer-landing.svg'
         width={702} height={512} alt="کلاسبن"/>
       </div>
     <div className="container">
     </div>
     </section>
        1`] = `
<div>
  <section
    class="bg-hero-pattern px-20 bg-no-repeat bg-center mt-5 xl:mt-20 xl:bg-left "
  >
    <div
      class="container flex flex-col-reverse items-center xl:flex-row "
    >
      <div
        class="flex flex-col gap-5 mt-12 pb-5 text-center   xl:text-right"
      >
        <h3
          class="text-xl dark:text-info xl:text-2xl"
        >
          خوش آومدی به ....
        </h3>
        <h1
          class="text-3xl font-black gradient lg:text-5xl xl:text-5xl "
        >
          مسیر صعود به قله های برنامه نویسی
        </h1>
        <p>
          هرجای مسیر برنامه نویسی که باشی با همراهی استادهای با تجربه کلاسبن می تونی بدون محدودیت به قله های بالاتر صعود کنی ،ما همیشه هواتو داریم
        </p>
        <div
          class="mt-5 flex gap-4"
        >
          <button
            class="btn btn-primary btn-lg"
            type="button"
          >
            دوره های ری اکت و نکست
          </button>
          <button
            class="btn btn-neutral btn-lg"
            type="button"
          >
            مشاوره برنامه نویسی
          </button>
        </div>
        <img
          alt=""
          class="grayscale mt-4 m-auto opacity-70 xl:m-0"
          data-nimg="1"
          decoding="async"
          height="39"
          loading="lazy"
          src="/_next/image?url=%2Fimages%2Fframeworks.png&w=828&q=75"
          srcset="/_next/image?url=%2Fimages%2Fframeworks.png&w=640&q=75 1x, /_next/image?url=%2Fimages%2Fframeworks.png&w=828&q=75 2x"
          style="color: transparent;"
          width="412"
        />
      </div>
      <img
        alt="کلاسبن"
        class=""
        data-nimg="1"
        decoding="async"
        height="512"
        loading="lazy"
        src="/images/programmer-landing.svg"
        style="color: transparent;"
        width="702"
      />
    </div>
    <div
      class="container"
    />
  </section>
</div>
`;
